src/Stratis.Bitcoin.Features.MemoryPool/MempoolRule.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Consensus;
+using Stratis.Bitcoin.Features.MemoryPool.Interfaces;
+
+namespace Stratis.Bitcoin.Features.MemoryPool
+{
+    public class MempoolRule : IMempoolRule
```

Comment:
This is ok but generally the specific implementation would be deciding what are its dependencies, otherwise you will have rules with dependencies they don't use.